.github/workflows/lint_python.yml

```
@@ -0,0 +1,18 @@
+name: lint_python
+on:
+  pull_request:
+  push:
+  #  branches: [master]
+jobs:
+  lint_python:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+      - uses: actions/setup-python@master
+      - run: pip install black codespell flake8 isort pytest
```

I would suggest to add these tooling dependencies to `requirements.txt` and then install everything in one command here:

```suggestion
      - run: pip install -r requirements.txt
```

The benefit of this approach is that it makes it easier to set up all the dependencies locally too.

---

Yes but... these are not runtime dependencies so there is no need for end users to install them.  They are only needed to test, not to run.  They create unnecessary code bloat foe end users.

---

`requirements.txt` in this project is somewhat incorrectly named and only contains development time dependencies. The library has no runtime dependencies as can be seen in `setup.py`. As such there's no concern to add dev tools to `requirements.txt`. If you wish, feel free to also rename the file to `requirements-dev.txt` while you're at it. Thanks!